src/lib/consensus/body_reference.ml

```diff
@@ -0,0 +1,53 @@
+(** This module defines how a block header refers to the body of its block.
```

Comment:
`consensus` seems like an odd place for this to live. Perhaps it would be better in `Mina_state`?

Reply:
An odd place, indeed

Here's why it's here though: https://github.com/MinaProtocol/mina/blob/e1f9b02b0cc6f93d56ce694e6ff336cd7834c932/src/lib/consensus/intf.ml#L44